app/scripts/directives/bindService.js

```diff
@@ -70,28 +74,45 @@ function BindService($filter,
     }
   };
 
-  ctrl.$onInit = function() {
-    ctrl.steps = [];
-    if (ctrl.target.kind === 'Instance') {
-      ctrl.steps.push({
-        id: 'applications',
-        label: 'Applications',
-        view: 'views/directives/bind-service/select-application.html'
-      });
-    }
-    else {
-      ctrl.steps.push({
-        id: 'services',
-        label: 'Services',
-        view: 'views/directives/bind-service/select-service.html'
-      });
-    }
-    ctrl.steps.push({
-      label: 'Results',
-      id: 'results',
-      view: 'views/directives/bind-service/results.html'
+  var showBind = function() {
+    ctrl.nextTitle = 'Bind';
+
+    validityWatcher = $scope.$watch("ctrl.selectionForm.$valid", function(isValid) {
+      ctrl.steps[0].valid = isValid;
     });
+  };
 
+  var showResults = function() {
+    if (validityWatcher) {
```

Comment:
The validity watcher is added to know when the form is valid, and we then update the next button enablement. It must be removed when we leave the page so we keep a handle to it and call it when the  page changes or the component is destroyed.